x-pack/test/security_solution_cypress/cypress/e2e/detection_response/rule_management/rules_table/rules_table_auto_refresh.cy.ts

```diff
@@ -51,73 +50,109 @@ describe(
+      expectAutoRefreshIsEnabled();
     });
 
-    it('should prevent table from rules refetch if any rule selected', () => {
-      mockGlobalClock();
-      visitWithoutDateRange(DETECTIONS_RULE_MANAGEMENT_URL);
+    describe('when enabled', () => {
+      beforeEach(() => {
+        mockGlobalClock();
+        visitWithoutDateRange(DETECTIONS_RULE_MANAGEMENT_URL);
 
-      expectNumberOfRules(RULES_MANAGEMENT_TABLE, NUM_OF_TEST_RULES);
+        expectNumberOfRules(RULES_MANAGEMENT_TABLE, NUM_OF_TEST_RULES);
+      });
+
+      it('refreshes rules after refresh interval has passed', () => {
+        cy.get(RULES_TABLE_AUTOREFRESH_INDICATOR).should('not.exist');
+        cy.tick(ONE_MINUTE_IN_MS);
+        cy.get(RULES_TABLE_AUTOREFRESH_INDICATOR).should('be.visible');
 
-      selectRulesByName(['Test rule 1']);
+        cy.contains(REFRESH_RULES_STATUS, 'Updated now');
+      });
 
-      // mock 1 minute passing to make sure refresh is not conducted
-      cy.get(RULES_TABLE_AUTOREFRESH_INDICATOR).should('not.exist');
-      cy.tick(DEFAULT_RULE_REFRESH_INTERVAL_VALUE);
-      cy.get(RULES_TABLE_AUTOREFRESH_INDICATOR).should('not.exist');
+      it('refreshes rules on window focus', () => {
+        cy.tick(ONE_MINUTE_IN_MS / 2);
 
-      // ensure rule is still selected
-      getRuleRow('Test rule 1').find(EUI_CHECKBOX).should('be.checked');
+        cy.window().trigger('blur');
+        cy.window().trigger('focus');
 
-      cy.get(REFRESH_RULES_STATUS).should('have.not.text', 'Updated now');
+        cy.contains(REFRESH_RULES_STATUS, 'Updated now');
+      });
     });
 
-    it('should disable auto refresh when any rule selected and enable it after rules unselected', () => {
-      visit(DETECTIONS_RULE_MANAGEMENT_URL);
+    describe('when disabled', () => {
+      beforeEach(() => {
+        mockGlobalClock();
+        visitWithoutDateRange(DETECTIONS_RULE_MANAGEMENT_URL);
+        // waitForPageToBeLoaded();
+        expectNumberOfRules(RULES_MANAGEMENT_TABLE, NUM_OF_TEST_RULES);
+        disableAutoRefresh();
+        cy.tick(ONE_MINUTE_IN_MS * 2); // Make sure enough time has passed to verify auto-refresh doesn't happen
+      });
 
-      expectNumberOfRules(RULES_MANAGEMENT_TABLE, NUM_OF_TEST_RULES);
+      it('does NOT refresh rules after refresh interval has passed', () => {
+        cy.contains(REFRESH_RULES_STATUS, 'Updated 2 minutes ago');
+      });
 
-      // check refresh settings if it's enabled before selecting
-      expectAutoRefreshIsEnabled();
+      it('does NOT refresh rules on window focus', () => {
+        cy.window().trigger('blur');
+        cy.window().trigger('focus');
 
-      selectAllRules();
+        // Without a delay here the following expectations pass even it shouldn't happen
+        // 'focus' event gets handled in async way so the status gets updated with some delay
+        // eslint-disable-next-line cypress/no-unnecessary-waiting
+        cy.wait(1000);
```

Comment:
Is this affecting just a single test case? Do we need to add `wait` to other cases too?

Reply:
It's required only for the test checking refresh doesn't happen on window focus. I've updated the comment to make clearer.